File: gems/decomplex/examples/python/implicit-control-flow.py
class FlowExample:
    def prepare(self): self.status = 1
    def validate(self): self.valid = self.status == 1
    def commit(self): self.done = self.valid
    def ok1(self): self.prepare(); self.validate(); self.commit()
    def ok2(self): self.prepare(); self.validate(); self.commit()
    def ok3(self): self.prepare(); self.validate(); self.commit()
    def ok4(self): self.prepare(); self.validate(); self.commit()
    def drift(self): self.validate(); self.prepare(); self.commit()

File: gems/decomplex/examples/php/implicit-control-flow.php
<?php
class FlowExample {
  public function prepare() { $this->status = READY; }
  public function validate() { $this->valid = $this->status == READY; }
  public function commit() { $this->done = $this->valid; }

  public function ok1() { $this->prepare(); $this->validate(); $this->commit(); }
  public function ok2() { $this->prepare(); $this->validate(); $this->commit(); }
  public function ok3() { $this->prepare(); $this->validate(); $this->commit(); }
  public function ok4() { $this->prepare(); $this->validate(); $this->commit(); }
  public function drift() { $this->validate(); $this->prepare(); $this->commit(); }
}
